scipy/signal/ltisys.py

```diff
@@ -555,18 +336,163 @@ def bode(self, w=None, n=100):
         Returns a 3-tuple containing arrays of frequencies [rad/s], magnitude
         [dB] and phase [deg]. See `scipy.signal.bode` for details.
 
-        Notes
-        -----
+        Examples
+        --------
+        >>> from scipy import signal
+        >>> import matplotlib.pyplot as plt
+
+        >>> sys = signal.TransferFunction([1], [1, 1])
+        >>> w, mag, phase = sys.bode()
+
+        >>> plt.figure()
+        >>> plt.semilogx(w, mag)    # Bode magnitude plot
+        >>> plt.figure()
+        >>> plt.semilogx(w, phase)  # Bode phase plot
+        >>> plt.show()
+
+        """
+        return bode(self, w=w, n=n)
+
+    def freqresp(self, w=None, n=10000):
+        """
+        Calculate the frequency response of a continuous-time system.
+
+        Returns a 2-tuple containing arrays of frequencies [rad/s] and
+        complex magnitude.
+        See `scipy.signal.freqresp` for details.
+
+        """
+        return freqresp(self, w=w, n=n)
+
+    def to_discrete(self, dt, method='zoh', alpha=None):
+        """Return a discretized version of the current system.
+
+        Parameters: See `cont2discrete` for details.
+
+        Returns
+        -------
+        sys: instance of `ltid`
+        """
+        raise NotImplementedError('to_discrete is not implemented for this '
+                                  'system class.')
+
+
+class ltid(LinearTimeInvariant):
+    """
+    Discrete-time linear time invariant system base class.
+
+    Parameters
+    ----------
+    *system: arguments
+        The `ltid` class can be instantiated with either 3, 4 or 5 arguments.
+        The following gives the number of arguments and the corresponding
+        subclass that is created:
+
+            * 3: `TransferFunction`:  (numerator, denominator)
+            * 4: `ZerosPolesGain`: (zeros, poles, gain)
+            * 5: `StateSpace`:  (A, B, C, D)
+
+        Each argument can be an array or a sequence.
+    dt: float
+        Optional, Sampling time [s] of the discrete-time systems. Defaults to
+        `True` (unspecified sampling time).
+
+    See Also
+    --------
+    ZerosPolesGain, StateSpace, TransferFunction, lti
+
+    Notes
+    -----
+    `ltid` instances do not exist directly. Instead, `ltid` creates an instance
+    of one of its subclasses: `StateSpace`, `TransferFunction` or
+    `ZerosPolesGain`.
+
+    Changing the value of properties that are not directly part of the current
+    system representation (such as the `zeros` of a `StateSpace` system) is
+    very inefficient and may lead to numerical inaccuracies.
+
+    If (numerator, denominator) is passed in for ``*system``, coefficients for
+    both the numerator and denominator should be specified in descending
+    exponent order (e.g., ``z^2 + 3s + 5`` would be represented as ``[1, 3,
+    5]``).
+
```

Comment:
Ah, I remembered wrongly. The reason for the `dt` keyword argument was that if it's not specified we default to a discrete-time system with `dt=True` otherwise. So it makes sense to keep it as it is currently.

A future PR of mine will try to deprecate being able to call `lti` and `ltid` directly anyways in favor of using the subclasses directly. The current way has too many access paths.